sync/src/chain.rs

```diff
@@ -1062,6 +1058,11 @@ impl ChainSync {
 				self.sync_peer(io, p, false);
 			}
 		}
+		if (self.state != SyncState::WaitingPeers && self.state != SyncState::SnapshotWaiting && self.state != SyncState::Waiting && self.state != SyncState::Idle)
+			&& !self.peers.values().any(|p| p.asking != PeerAsking::Nothing && p.block_set != Some(BlockSet::OldBlocks) && p.can_sync()) {
```

Comment:
Makes more sense to check if all peers are idle after they have been given new tasks and not before